module-bsp/board/rt1051/puretx/bsp/battery_charger/battery_charger.cpp

```diff
@@ -307,52 +324,113 @@ namespace bsp::battery_charger
 
         batteryRetval storeConfiguration()
         {
-            LOG_INFO("Storing fuel gauge configuration...");
             std::ofstream file(cfgFile.c_str(), std::ios::binary | std::ios::out);
             if (!file.is_open()) {
                 LOG_WARN("Configuration file [%s] could not be opened.", cfgFile.c_str());
                 return batteryRetval::ChargerError;
             }
 
+            std::uint16_t checksum = 0;
+            std::vector<std::uint16_t> dataToStore;
             for (unsigned int i = 0; i < registersToStore; ++i) {
                 auto regVal = fuelGaugeRead(static_cast<Registers>(i));
                 if (regVal.first != kStatus_Success) {
                     LOG_ERROR("Reading register 0x%x failed.", i);
+                    file.close();
                     return batteryRetval::ChargerError;
                 }
+                checksum ^= regVal.second;
```

Comment:
I'd be better to use specific library to calculate crc:)

Reply:
Done